### ISSUE TITLE
Feature Request: make a specific replication job unstoppable by the scheduler

### ISSUE THREAD

**Issue by skeyby** [apache/couchdb#1720]
As a companion to Feature Request #1714 , another very useful toggle would be a property in the Replication DOC to force a specific replication job to avoid being paused by replication scheduler.

In our scenario we have a "main" replication Job that keeps a central machine aligned with the server with all the users db, what we'd like to obtain is defining the main replication job as "unstoppable" while having N-replication jobs for users DBs handled by the replication scheduler.

I think that could be implemented in a fairly easy way by setting a property that would instruct the  rotate_jobs() + longest_running function to ignore a replication task.

(I see there already are a few ideas close to this one: #1412, #1257...)

**Comment by skeyby** [comment]
_(note: obviously I can setup the main replication as a push replication from the central server to the users db one to work around current behaviour, I know that, but the basic feature request remain...)_

**Comment by nickva** [comment]
Another way to set this up is to make the replication a `normal` (non-continuous) replication. Normal replications have priority and unless an operator explicitly dials down `max_jobs` parameter or restarts a replication node (server).

The reason for this behavior is that traditionally (before the scheduling replicator was created) a `normal` replication replicates a snapshot of the database at particular time. So say you replicate from source S to target T. Then after the replication starts, you add a document do S. That document won't appear in T if replication is `normal`. If replication is `continuous` then it will. 

So basically if you have a bunch of continuous replications but want to make one replication unstoppable, have your user code periodically create normal replications from source to target. Monitor when the replication finishes and re-create it.

In some cases it might even be helpful to use the `_replicate` endpoint which doesn't involve creating a replication document, which is ok if you create it and monitor it programmatically.

**Comment by skeyby** [comment]
You're right but when you start a new full replication the whole db has to be scanned for already present and updated docs, when a DB has many documents that would be an huge overkill. The proposal work-around this issue by keeping the sequence number at hand.

Furthermore the whole point is to keep the "master" replication flowing in a _near-realtime-fashion_, if I have to restart it every, let's say, 10 minutes then the currently stop-and-start by the scheduler it's already enough.

**Comment by kocolosk** [comment]
@nickva I think the new [Fair Share](https://docs.couchdb.org/en/stable/replication/replicator.html#fair-share-job-scheduling) replicator feature allows for this capability?

1. Create an `always-running/_replicator` DB
2. Load replication docs for high priority jobs into this DB instead of the main one
3. Ensure `[replicator.shares]` allocates enough shares to prioritize the `always-running` jobs

The last step is probably not required if the number of jobs in that priority DB is small, as the algorithm will default to sharing resources fairly at the level of the replicator DBs.

I suppose it's not a hard guarantee that the priority jobs will *always* be running, but does it effectively deliver the same outcome?

**Comment by nickva** [comment]
@kocolosk that would work, yeah. Can set a high number of shares (1000 max) for `_replicator/always-running` and those jobs should get a high priority to always run in front of jobs from say `_replicator` or `_replicator/user-jobs`. To lower the chance even more for other dbs running, can explicitly grant them the minimum  - `1` share

We could also add a small enhancement to completely disable some replication db (assign them 0 shares, I think now 1 is the minimum) that would guarantee that they never get picked.

I am inclined to close this for now as fixed and if after evaluating the new Fair Share is not adequate, to re-open it.
